cli/pkg/kctrl/cmd/package/init.go

```diff
@@ -64,54 +64,55 @@ func (o *InitOptions) Run() error {
 		}
 	}
 
+	o.ui.PrintInformationalText("\nWelcome! Before we start, do install the latest Carvel suite of tools, specifically ytt, imgpkg, vendir and kbld as these will be used by kctrl.\n")
+	o.ui.PrintHeaderText("\nBasic Information")
+
 	pkgBuild, err := o.newOrExistingPackageBuild()
 	if err != nil {
 		return err
 	}
 
 	pkg, pkgMetadata, pkgInstall, err := o.newOrExistingPackageResources()
 
-	o.ui.PrintInformationalText("\nWelcome! Before we start, do install the latest Carvel suite of tools, specifically ytt, imgpkg, vendir and kbld as these will be used by kctrl.\n")
-
-	o.ui.PrintHeaderText("\nBasic Information")
-
 	pkgRefName, err := o.readPackageRefName(pkgMetadata.Name)
 	if err != nil {
 		return err
 	}
 
-	// TODO: @praveenrewar Can this be don any better?
-	pkgMetadata.Name = pkgRefName
-	pkg.Spec.RefName = pkgRefName
-	pkgMetadata.Spec.DisplayName = strings.Split(pkgRefName, ".")[0]
+	o.updateAndDefaultPackageResources(pkgRefName, pkgMetadata, pkg)
 
-	shortDesc := pkgMetadata.Spec.ShortDescription
-	if len(shortDesc) == 0 {
-		pkgMetadata.Spec.ShortDescription = pkgRefName
+	err = o.savePackageResources(pkg, pkgMetadata, pkgInstall)
+	if err != nil {
+		return err
 	}
-
-	longDesc := pkgMetadata.Spec.LongDescription
-	if len(longDesc) == 0 {
-		pkgMetadata.Spec.LongDescription = pkgRefName
+	err = pkgBuild.Save()
+	if err != nil {
+		return err
 	}
 
-	err = o.SavePackageResources(pkg, pkgMetadata, pkgInstall)
+	fetchMode, err := sources.NewSource(o.ui, pkgBuild).Configure()
 	if err != nil {
 		return err
 	}
-	pkgBuild.Save()
+
+	err = sources.NewVendirRunner(o.ui).Sync(fetchMode)
 	if err != nil {
 		return err
 	}
 
-	// TODO: @praveenrewar Remove the step part and use only relevant code from Fetch
-	appCreateStep := appinit.NewCreateStep(o.ui, pkgBuild, o.logger, o.depsFactory, false)
-	err = interfaces.Run(appCreateStep)
+	err = sources.NewTemplateConfiguration(o.ui, pkgBuild).Configure(fetchMode)
 	if err != nil {
 		return err
 	}
 
-	pkgBuild.Save()
+	appSpec := pkgBuild.GetAppSpec()
+	if appSpec.Deploy == nil {
+		appSpec.Deploy = []kcv1alpha1.AppDeploy{{Kapp: &kcv1alpha1.AppDeployKapp{}}}
+	}
+	pkgBuild.SetAppSpec(appSpec)
```

Comment:
Maybe all of this should be under on function? (It seems like we are just setting deploy value to if it's nil).

Reply:
Done!